src/Project.js

```diff
@@ -578,21 +578,55 @@ class Project extends DirItem {
         // source file in order to make it easier for the engineer to fix all the
         // problems in the source file sequentially.
         results.sort(ResultComparator);
-
+        let resultFull;
         if (results) {
+            results.forEach(result => {
+                if (result.severity === "error") {
+                    this.resultStats.errors++;
+                } else if (result.severity === "warning") {
+                    this.resultStats.warnings++;
+                } else {
+                    this.resultStats.suggestions++;
+                }
+            });
+        }
+        const score = this.getScore();
+
+        if (this.options.opt["max-errors"]) {
+            exitValue = this.resultStats.errors > this.options.opt["max-errors"] ? 2 : 0;
+        } else if (this.options.opt["max-warnings"]) {
+            exitValue = this.resultStats.warnings > this.options.opt["max-warnings"] ? 1 : 0;
+        } else if (this.options.opt["max-suggestions"]) {
+            exitValue = this.resultStats.suggestions > this.options.opt["max-suggestions"] ? 1 : 0;
+        } else if (this.options.opt["min-score"]) {
+            exitValue = score < this.options.opt["min-score"] ? 2 : 0;
+        } else if (this.options.errorsOnly) {
+            exitValue = this.resultStats.errors > 0 ? 2 : 0;
+        } else {
+            exitValue = this.resultStats.errors > 0 ? 2 : ((this.resultStats.warnings > 0) ? 1 : 0);
+        }
+
+        if (typeof (this.formatter.formatOutput) === "function") {
+            resultFull = this.formatter.formatOutput({
+                name: this.project.name,
+                fileStats: this.fileStats,
+                resultStats: this.resultStats,
+                results: results,
+                score: score,
+                time: totalTime,
+                errorOnly : this.options.opt.errorsOnly
+            });
+        } else {
             results.forEach(result => {
```

Comment:
This "else" section needs to set the resultFull string if this.options.opt.output is set, and not do any output to the logger. If it is not set, then output everything on the fly.

Reply:
I update it.